### PR TITLE
adjust encryption bit while tracing outgoing wireguard-encrypted packets

### PR DESCRIPTION
**Pull request by smagnani96**
With this PR we just set the tracing encryption bit rather than overwriting trace reasons, causing potential loss of information. Offending commit hash 8a6738765ea6d23de4dbf63e2c37048adea7e69b, PR https://github.com/cilium/cilium/pull/35183.

**_NOTE_**:  backport shouldn't be needed since the offending commit made it to pre-release only https://github.com/cilium/cilium/releases/tag/v1.17.0-pre.2.

```release-note
Fix incorrect trace reason for egress packets when WireGuard is used with Host Firewall.
```

### CODE REVIEW

File: bpf/lib/nodeport_egress.h [hunk 1]
```diff
@@ -100,13 +100,19 @@ int tail_handle_snat_fwd_ipv6(struct __ctx_buff *ctx)
...
 #ifdef IS_BPF_OVERLAY
 	obs_point = TRACE_TO_OVERLAY;
 #else
 	obs_point = TRACE_TO_NETWORK;
+
+#ifdef ENABLE_WIREGUARD
+	if ((ctx->mark & MARK_MAGIC_WG_ENCRYPTED) == MARK_MAGIC_WG_ENCRYPTED)
+		encryption_bit = TRACE_REASON_ENCRYPTED;
+#endif /* ENABLE_WIREGUARD */
 #endif
```
Comment (julianwiedmann) [inline bpf/lib/nodeport_egress.h]: Before we start adding this check to all the deep-down NAT flows, I think it makes sense to reconsider whether our WG traffic should even go down that path. Or be excluded [here](https://github.com/cilium/cilium/blob/0e4bc5e39ba2927f8c829c7f9f0626cd5b0f2e05/bpf/bpf_host.c#L1617).

We clearly don't care about the RevDNAT case for service replies. So it comes down to understanding the SNAT case a bit better.

**Comment by smagnani96** [comment]
> Before we start adding this check to all the deep-down NAT flows, I think it makes sense to reconsider whether our WG traffic should even go down that path. Or be excluded [here](https://github.com/cilium/cilium/blob/0e4bc5e39ba2927f8c829c7f9f0626cd5b0f2e05/bpf/bpf_host.c#L1617).
 We clearly don't care about the RevDNAT case for service replies. So it comes down to understanding the SNAT case a bit better.

As discussed offline, I just dropped the respective commits and left only the first fix.
The concern with SNAT will be tackled in a separate Issue/PR.

**Review by julianwiedmann — APPROVED** [review]
Thank you!

**Pull request merged**